src/core/mount.c

```diff
@@ -270,15 +271,28 @@ static int update_parameters_proc_self_mountinfo(
                 const char *fstype) {
 
         MountParameters *p;
+        _cleanup_free_ char *what_escaped = NULL, *options_escaped = NULL;
         int r, q, w;
 
         p = &m->parameters_proc_self_mountinfo;
 
-        r = free_and_strdup(&p->what, what);
+        if (what) {
+                what_escaped = utf8_escape_invalid(what);
+                if (!what_escaped)
+                        return -ENOMEM;
+        }
+
+        r = free_and_replace(p->what, what_escaped);
```

Comment:
free_and_replace() does not compare with the previous string, and always return 0.